DependencyInjection/Compiler/DescriptionEnhancerPass.php

```diff
@@ -59,11 +70,14 @@ public function process(ContainerBuilder $container)
             ));
         }
 
-        $inactiveEnhancers = array_diff($enhancerNames, $enabledEnhancers);
-        foreach ($inactiveEnhancers as $inactiveEnhancer) {
-            $container->removeDefinition((string) $enhancers[$inactiveEnhancer]);
-            unset($enhancers[$inactiveEnhancer]);
-        }
+        // sort enhancers, higher = more priority
+        usort($enhancers, function ($a, $b) {
+            return -strcmp($a[0], $b[0]);
```

Comment:
Good point :)

On Thu, Jun 23, 2016 at 07:24:39AM -0700, David Buchmann wrote:
>    In [1]DependencyInjection/Compiler/DescriptionEnhancerPass.php:
> 
>  > @@ -59,11 +70,14 @@ public function process(ContainerBuilder $container)
>  >              ));
>  >          }
>  >
>  > -        $inactiveEnhancers = array_diff($enhancerNames, $enabledEnhancers);
>  > -        foreach ($inactiveEnhancers as $inactiveEnhancer) {
>  > -            $container->removeDefinition((string) $enhancers[$inactiveEnhancer]);
>  > -            unset($enhancers[$inactiveEnhancer]);
>  > -        }
>  > +        // sort enhancers, higher = more priority
>  > +        usort($enhancers, function ($a, $b) {
>  > +            return -strcmp($a[0], $b[0]);
> 
>    what about strcmp($b[0], $a[0])? the - in front of strcmp confused me for
>    a bit...
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly, [2]view it on GitHub, or [3]mute the thread.
> 
>    Reverse link: [4]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/symfony-cmf/resource-bundle/pull/34#discussion_r68243755
>    2. https://github.com/symfony-cmf/resource-bundle/pull/34/files/2c1a09f0964f149a4f23558bba0773e999fbd4d9#r68243755
>    3. https://github.com/notifications/unsubscribe/AAgZcbEaYgU8nTL2B7Ie55vcdbv4JMAIks5qOpcngaJpZM4I6bpJ
>    4. https://github.com/symfony-cmf/resource-bundle/pull/34/files/2c1a09f0964f149a4f23558bba0773e999fbd4d9#r68243755